spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -494,6 +497,62 @@ public void testRecordAck() throws Exception {
 		logger.info("Stop record ack");
 	}
 
+	@Test
+	public void testBatchAck() throws Exception {
+		logger.info("Start batch ack");
+
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		ProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(topic7);
+		template.sendDefault(0, 0, "foo");
+		template.sendDefault(0, 0, "baz");
+		template.sendDefault(1, 0, "bar");
+		template.sendDefault(1, 0, "qux");
+		template.flush();
+
+		Map<String, Object> props = KafkaTestUtils.consumerProps("test6", "false", embeddedKafka);
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+		DefaultKafkaConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<>(props);
+		ContainerProperties containerProps = new ContainerProperties(topic7);
+		containerProps.setMessageListener((MessageListener<Integer, String>) message -> {
+			logger.info("batch ack: " + message);
+		});
+		containerProps.setSyncCommits(true);
+		containerProps.setAckMode(AckMode.BATCH);
+		containerProps.setPollTimeout(10000);
+		containerProps.setAckOnError(false);
+		KafkaMessageListenerContainer<Integer, String> container = new KafkaMessageListenerContainer<>(cf,
+				containerProps);
+		container.setBeanName("testBatchAcks");
+		container.start();
+		Consumer<?, ?> containerConsumer = spyOnConsumer(container);
+		final CountDownLatch latch = new CountDownLatch(2);
+		willAnswer(invocation -> {
+
+			@SuppressWarnings({ "unchecked" })
+			Map<TopicPartition, OffsetAndMetadata> map = (Map<TopicPartition, OffsetAndMetadata>) invocation
+					.getArguments()[0];
+			for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
+				if (entry.getValue().offset() == 2) {
+					latch.countDown();
+				}
+			}
+			return invocation.callRealMethod();
+
+		}).given(containerConsumer)
+				.commitSync(any());
+
+		assertThat(latch.await(60, TimeUnit.SECONDS)).isTrue();
```

Comment:
I think we need to verify the `commit()` is called within 10 seconds of the first message received.